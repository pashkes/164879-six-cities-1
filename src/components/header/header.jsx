import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {Link} from "react-router-dom";

import Constants from "./../../constants";
import {getUserData, getAuthorizationStatus} from "./../../reducer/user/selectors";

export const Header = (props) => {
  const {isAuthorization, userData} = props;
  return (
    <header className="header">
      <div className="container">
        <div className="header__wrapper">
          <div className="header__left">
            <Link to="/" className="header__logo-link header__logo-link--active">
              <img className="header__logo" src="img/logo.svg" alt="6 cities logo" width="81" height="41"/>
            </Link>
          </div>
          <nav className="header__nav">
            <ul className="header__nav-list">
              <li className="header__nav-item user">
                <Link to={isAuthorization ? `./favorites` : `/login`} className="header__nav-link header__nav-link--profile">
                  <div
                    className="header__avatar-wrapper user__avatar-wrapper"
                    style={isAuthorization ? {backgroundImage: `url(${Constants.BASE_URL}${userData.avatar})`} : {}}>
                  </div>
                  <span className="header__user-name user__name">{isAuthorization ? userData.email : `Sign In`}</span>
                </Link>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
  );
}
;

Header.propTypes = {
  isAuthorization: PropTypes.bool.isRequired,
  userData: PropTypes.object.isRequired,
};


const mapStateToProps = (state) => ({
  userData: getUserData(state),
  isAuthorization: getAuthorizationStatus(state)
});

export default connect(mapStateToProps)(Header);
