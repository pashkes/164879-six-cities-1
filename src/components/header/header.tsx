import * as React from "react";
import {connect} from "react-redux";
import {Link} from "react-router-dom";

import Constants, {Page} from "./../../constants";
import {getUserData, getAuthorizationStatus} from "./../../reducer/user/selectors";

interface Props {
  isAuthRequire: boolean,
  userData: {
    avatar: string,
    email: string,
    id: number,
    isPro: boolean,
    name: string,
  },
}

export const Header: React.FunctionComponent<Props> = (props) => {
  const {isAuthRequire, userData} = props;

  return (
    <header className="header">
      <div className="container">
        <div className="header__wrapper">
          <div className="header__left">
            <Link to="/" className="header__logo-link header__logo-link--active">
              <img className="header__logo" src="img/logo.svg" alt="6 cities logo" width="81" height="41"/>
            </Link>
          </div>
          <nav className="header__nav">
            <ul className="header__nav-list">
              <li className="header__nav-item user">
                <Link to={isAuthRequire ? Page.LOGIN : Page.FAVORITES}
                      className="header__nav-link header__nav-link--profile">
                  <div
                    className="header__avatar-wrapper user__avatar-wrapper"
                    style={isAuthRequire ? {} : {backgroundImage: `url(${Constants.BASE_URL}${userData.avatar})`}}>
                  </div>
                  <span className="header__user-name user__name">{isAuthRequire ? `Sign In` : userData.email}</span>
                </Link>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
  );
};


const mapStateToProps = (state) => ({
  userData: getUserData(state),
  isAuthRequire: getAuthorizationStatus(state)
});

export default connect(mapStateToProps)(Header);
