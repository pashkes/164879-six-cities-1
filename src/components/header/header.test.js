import React from "react";
import {Header} from "./header.jsx";
import {shallow, configure} from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({adapter: new Adapter()});

it(`Header renders correctly`, () => {
  const cards = shallow(<Header
    isAuthorization={false}
    userData={{
      avatar: ``,
      email: ``,
    }}
    onClickLogOut={jest.fn()}
    logOut={jest.fn()}
  />);
  expect(cards).toMatchSnapshot();
});
