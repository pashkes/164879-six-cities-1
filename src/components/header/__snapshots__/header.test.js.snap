// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <Link
          className="header__logo-link header__logo-link--active"
          to="/"
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="img/logo.svg"
            width="81"
          />
        </Link>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <Link
              className="header__nav-link header__nav-link--profile"
              to="/favorites"
            >
              <div
                className="header__avatar-wrapper user__avatar-wrapper"
                style={
                  Object {
                    "backgroundImage": "url(https://es31-server.appspot.com/six-cities)",
                  }
                }
              />
              <span
                className="header__user-name user__name"
              />
            </Link>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;
